wandb/sdk/launch/agent/agent.py

```diff
@@ -328,6 +328,25 @@ async def update_status(self, status: str) -> None:
         if not update_ret["success"]:
             wandb.termerror(f"{LOG_PREFIX}Failed to update agent status to {status}")
 
+    def _check_run_exists_and_inited(
+        self, entity: str, project: str, run_id: str, rqi_id: str
+    ) -> bool:
+        """Checks the stateof the run to ensure it has been inited. Note this will not behave well with resuming."""
```

Comment:
This is quite ominous.. Can we detect/handle the resumed case in any way?

Reply:
We can on future releases by checking the RunQueueItemID field in the run's config. But even that will have ill-behaved race conditions. We could inject rqi ID keys with boolean values into the config which should work.

I can add a TODO, this new behavior is not worse than the original, but I did want to call it out.